apps/studio/data/analytics/infra-monitoring-queries.ts

```diff
@@ -12,18 +30,83 @@ export function useInfraMonitoringQueries(
   data: any,
   isVisible: boolean
 ) {
-  return attributes.map((attribute) =>
-    // eslint-disable-next-line react-hooks/rules-of-hooks
-    useInfraMonitoringQuery(
-      {
-        projectRef: ref as string,
-        attribute,
-        startDate,
-        endDate,
-        interval,
-        databaseIdentifier,
-      },
-      { enabled: data === undefined && isVisible }
-    )
+  const shouldFetch = data === undefined && isVisible
+  const hasAttributes = attributes.length > 0
+  const isSingleAttribute = attributes.length === 1
+  const primaryAttribute = attributes[0] ?? DEFAULT_ATTRIBUTE
+
+  const singleAttributeQuery = useInfraMonitoringQuery(
+    {
+      projectRef: ref as string,
+      attribute: primaryAttribute,
+      startDate,
+      endDate,
+      interval,
+      databaseIdentifier,
+    },
+    { enabled: shouldFetch && isSingleAttribute && hasAttributes }
   )
+
+  const multiQuery = useInfraMonitoringAttributesQuery(
+    {
+      projectRef: ref as string,
+      attributes,
+      startDate,
+      endDate,
+      interval,
+      databaseIdentifier,
+    },
+    { enabled: shouldFetch && attributes.length > 1 }
+  )
+
+  const seriesByAttribute = useMemo(() => {
+    if (!multiQuery.data) return undefined
+    return mapMultiResponseToAnalyticsData(multiQuery.data, attributes)
+  }, [multiQuery.data, attributes])
+
+  if (!hasAttributes) {
+    return []
+  }
+
+  if (isSingleAttribute) {
+    return [singleAttributeQuery]
+  }
+
+  return attributes.map<InfraQueryResult>((attribute) => ({
+    data: seriesByAttribute?.[attribute],
+    error: multiQuery.error,
+    isError: multiQuery.isError,
+    isFetching: multiQuery.isFetching,
+    isLoading: multiQuery.isLoading,
+    status: multiQuery.status,
+  }))
+}
+
+function mapMultiResponseToAnalyticsData(
+  response: InfraMonitoringMultiData,
+  attributes: InfraMonitoringAttribute[],
+  dateFormat: string = DEFAULT_DATE_FORMAT
+) {
+  return attributes.reduce<Record<string, AnalyticsData>>((acc, attribute) => {
+    const metadata = response.series?.[attribute]
+    if (!metadata) return acc
+
+    const dataPoints = response.data.map((point) => {
+      const value = point.values?.[attribute]
+      return {
+        period_start: point.period_start,
+        periodStartFormatted: dayjs(point.period_start).format(dateFormat),
+        [attribute]: value === undefined ? 0 : Number(value),
+      }
+    })
+
+    acc[attribute] = {
+      data: dataPoints,
+      format: metadata.format,
+      total: metadata.total,
+      yAxisLimit: metadata.yAxisLimit,
+    }
+
+    return acc
+  }, {})
```

Comment:
Thoughts on writing tests around this method alone? Might be useful to bake the contract in

Reply:
710f5e0c3c29721bae2ec7ca999a1cb9b9cab1df